snuba/datasets/schemas/tables.py

```diff
@@ -15,16 +16,24 @@ class TableSource(RelationalSource):
     datamodel.
     """
 
-    def __init__(self, table_name: str, columns: ColumnSet) -> None:
+    def __init__(self,
+        table_name: str,
+        columns: ColumnSet,
+        mandatory_conditions: Sequence[Condition],
```

Comment:
Would it make sense to make this `Optional`, and set the default as an empty list if `None` is provided? (Note: I wrote this before reading the remainder of this file, and I think that provides evidence that it should.)

This would also remove some noise from a couple of the dataset definitions, and also allow you to revert some of the test changes that instantiated this value.

Reply:
Yeah, I went back and forth a bit on those two approaches. I am ok making it optional.